Update Cobl

charabaruk opened the pull request.

charabaruk committed:
Add link for Cobl Main.esm

charabaruk committed:
Add file anchor for Cobl

Make it easier to list Cobl as a requirement for mods that need it. Use
the new anchor in place of 'Cobl Main.esm' directly.

sibir-ine commented on masterlist.yaml:
```diff
@@ -1449,6 +1453,9 @@ plugins:
       - link: 'https://www.nexusmods.com/oblivion/mods/21104/'
         name: 'Common Oblivion (Cobl)'
   - name: 'Cobl Main.esm'
+    url:
+      - link: 'https://www.nexusmods.com/oblivion/mods/21104/'
+        name: 'Common Oblivion (Cobl)'
```
```suggestion
```

Since the location is already in a regex plugin just above this one, I'd prefer to just expand it. You can do so by changing `.*(Cobl).*\.esp` to `.*Cobl.*\.es[mp]` as that would apply the location to ESMs as well (the `()` doesn't actually do anything there & can just be removed).

charabaruk replied:
Not sure on that one; I think the existing regex is actually too hungry to begin with since it captures _any_ plugin that contains "Cobl" in the filename even if they are from different mod projects (e.g. my alchemy station and Sutch Coblized projects, but also pretty much any other pure Coblizer project). Most of the actual Cobl plugins all start with "Cobl " -- only some dev patches, _Salmo the Baker Cobl_, and _Item Interchange_ plugins in the Cobl download don't (and the last are already defined in their own section anyway). `Cobl.*\.es[mp]` might be a better regex to limit the link to just the actual Cobl files, possibly adding another item for `Salmo the Baker, Cobl.*\.esp` to catch those too. Alternatively, `(Salmo the Baker, )?Cobl.*\.es[mp]`?

sibir-ine replied:
`(Salmo the Baker, )?Cobl.*\.es[mp]` is fine. Locations are primarily for maintainers, so, as long as plugins from the same page are all grouped together, plugins from separate pages are separated, & there's a location somewhere in each grouping, it should be clear which plugins come from where.

sibir-ine commented on masterlist.yaml:
```diff
@@ -280,6 +280,10 @@ common:
     display: '[SoundCommands 4.0](https://www.nexusmods.com/oblivion/mods/16288)'
 
   # Mod Files
+  - &Cobl
+    name: 'Cobl Main.esm'
+    display: '[Common Oblivion (Cobl)](https://www.nexusmods.com/oblivion/mods/21104/)'
+
   - &BetterCities
     name: 'Better Cities Resources.esm'
     display: '[Better Cities](https://www.nexusmods.com/oblivion/mods/16513)'
```
```suggestion
  - &BetterCities
    name: 'Better Cities Resources.esm'
    display: '[Better Cities](https://www.nexusmods.com/oblivion/mods/16513)'

  - &Cobl
    name: 'Cobl Main.esm'
    display: '[Common Oblivion (Cobl)](https://www.nexusmods.com/oblivion/mods/21104/)'
```

These should be sorted alphabetically.

sibir-ine commented on masterlist.yaml:
```diff
@@ -28851,7 +28858,7 @@ plugins:
     url: [ 'https://www.nexusmods.com/oblivion/mods/10671/' ]
     inc:
```
```suggestion
    inc:
      - *Cobl
```

Aliases should be sorted above non-aliases for consistency.

sibir-ine commented on masterlist.yaml:
```diff
@@ -28851,7 +28858,7 @@ plugins:
     inc:
       - 'Mart''s Monster Mod - Diverse Runeskulls Loot G&GD.esp'
-      - 'Cobl Main.esm'
+      - *Cobl
```
```suggestion
```

(I had to split this suggestion up for GH to accept it unfortunately).

sibir-ine commented on masterlist.yaml:
```diff
@@ -30050,7 +30057,7 @@ plugins:
     req:
```
```suggestion
    req:
      - *Cobl
```

sibir-ine commented on masterlist.yaml:
```diff
@@ -30050,7 +30057,7 @@ plugins:
       - *SilentVoice
       - 'DLCShiveringIsles.esp'
-      - 'Cobl Main.esm'
+      - *Cobl
```
```suggestion
```

charabaruk committed:
resort Cobl anchor and aliases

Put the anchor after the one for Better Cities so that they are in
alphabetical order. Place aliases ahead of non-aliases and in
alphabetical order where amongst fellow aliases.

charabaruk committed:
Update regex for matching Cobl files

This will catch all actual Cobl files (other than some specific dev
patches) without applying the Cobl link to patches from other projects
that add Cobl features.
